cromwell-drs-localizer/src/main/scala/drs/localizer/CommandLineParser.scala

```diff
@@ -0,0 +1,80 @@
+package drs.localizer
+
+import common.util.VersionUtil
+import drs.localizer.CommandLineParser.AccessTokenStrategy._
+import drs.localizer.CommandLineParser.Usage
+
+
+class CommandLineParser extends scopt.OptionParser[CommandLineArguments](Usage) {
+  lazy val localizerVersion: String = VersionUtil.getVersion("cromwell-drs-localizer")
+
+  private def commonArguments = List(
+    arg[String]("drs-object-id").text("DRS object ID").required().
+      action((s, c) =>
+        c.copy(drsObject = Option(s))),
+    arg[String]("container-path").text("Container path").required().
+      action((s, c) =>
+        c.copy(containerPath = Option(s))),
+  )
+
+  version("version")
+
+  help("help").text("Cromwell DRS Localizer")
+
+  head("cromwell-drs-localizer", localizerVersion)
+
+  cmd("azure").
+    text("Localize DRS file using Azure UAMI / B2C access token strategy").
+    action((s, c) =>
+      c.copy(accessTokenStrategy = Option("azure"))).
+    children(commonArguments ++ List(
+      opt[String]('v', "vault-name").text("Azure vault name").
+        action((s, c) =>
+          c.copy(azureVaultName = Option(s))),
+      opt[String]('s', "secret-name").text("Azure secret name").
+        action((s, c) =>
+          c.copy(azureSecretName = Option(s))),
+      opt[String]('i', "identity-client-id").text("Azure identity client id").
+        action((s, c) =>
+          c.copy(azureIdentityClientId = Option(s))),
+    ): _*)
+
+  cmd("google").
+    text("Localize DRS file using Google Application Default Credentials access token strategy").
+    action((s, c) =>
+      c.copy(accessTokenStrategy = Option("google"))).
+    children(commonArguments ++ List(
+      opt[String]('r', "requester-pays-project").text("Google requester pays project name").
+        action((s, c) =>
+          c.copy(googleRequesterPaysProject = Option(s))),
+    ): _*)
+}
+
+object CommandLineParser {
+  /**
+   * These access token strategies are named simplistically as there is currently only one access token strategy being
+   * used for each of these cloud vendors. But it is certainly possible that multiple strategies could come into use
+   * for a particular vendor, in which case the names may need to become more specific for disambiguation.
+   */
+  object AccessTokenStrategy {
+    val Azure = "azure"
+    val Google = "google"
+  }
+
+  val Usage =
+    s"""
+Usage:
+    java -jar /path/to/localizer.jar $Azure drs://provider/object /local/path/to/file.txt [--vault-name <name>] [--secret-name <name>] [--identity-client-id <id>]
+OR
+    java -jar /path/to/localizer.jar $Google drs://provider/object /local/path/to/file.txt [--requester-pays-project <project>]
```

Comment:
I'm worried that this changes how the localizer is invoked. That may be fine for Cromwell if we also update it at the same time, but there could be others using this utility, including (soon) CromwellOnAzure.

Can the new arguments be done in such a way as to keep the existing positional arguments as-is?

Reply:
We could keep the positional arguments the same but if we want to support the old two arg invocation syntax we'd have to assume a default Google access token strategy. I'd really prefer not to do that as we bring Azure and GCP to parity.

I don't believe the Cromwell DRS Localizer is used anywhere outside of Terra at the moment, and any references in Cromwell config would be to pinned versions anwyay.

Reply:
You make a good point about pinned versions in Cromwell configs. I also don't see the harm in assuming Google as the default if we don't see command line options to indicate that we should use Azure.

Ultimately, if/when we make a breaking change, I'd like to make sure it shows up in the release notes.

Reply:
Since the Azure auth is still special-case for the demo, I feel more comfortable defaulting to Google than I normally would. Would also be nice to minimize future changes needed for this code, though.